src/main/java/org/broadinstitute/hellbender/tools/spark/sv/ContigAligner.java

```diff
@@ -0,0 +1,272 @@
+        }
+        return quals;
+    }
+
+    public void close() {
+        bwaMem.dispose();
+    }
+
+    /**
+     * Holds information about a split alignment of a contig, which may represent an SV breakpoint. Each AssembledBreakpoint
+     * represents the junction on the contig of two aligned regions. For example, if a contig aligns to three different regions
+     * of the genome (with one primary and two supplementary alignment records), there will be two AssembledBreakpoint
+     * objects created, one to represent each junction between alignment regions:
+     *
+     * Example Contig:
+     * ACTGACTGCACTGACTGCACTGACTGCACTGACTGCACTGACTGCACTGACTGCACTGACTGCACTGACTGCACTGACTGCACTGACTGCACTGACTGCACTGACTG
+     * Alignment regions:
+     * |---------1:100-200------------|
+     *                                |-----------2:100-200------------------|
+     *                                                                       |----------3:100-200-----------------|
+     * Assmbled breakpoints:
+     * 1) links 1:100-200 to 2:100-200
+     * 2) links 2:100-200 to 3:100-200
+     *
+     * Inserted sequence contains portions of the contig that are aligned to neither region, and therefore may be inserted in
+     * the sample. For example, a translocation breakpoint with a microinsertion:
+     *
+     * Contig:
+     * ACTGACTGACTGACTGACTGACTGACTGACTGACTGACTGACTG
+     * Alignment regions:
+     * |-----1:100-200-------|
+     *                         |-----2:100-200-----|
+     * Inserted sequence:
+     *  GA
+     *
+     * Homology represents ambiguity about the exact location of the breakpoint. For example, in this case one alignment
+     * region ends with "AC" and the next begins with AC, so we don't know if the AC truly belongs with the first or
+     * second alignment region.
+     *
+     * Contig:
+     * ACTGACTGACTGACTGACTGACTGACTGACTGACTGACTGACTG
+     * Alignment regions:
+     * |-----1:100-200-------|
+     *                    |-----2:100-200----------|
+     * Homology:
+     *  AC
+     */
+    static class AssembledBreakpoint {
+        String contigId;
+        AlignmentRegion region1;
+        AlignmentRegion region2;
+        String insertedSequence;
+        String homology;
+
+        @Override
+        public String toString() {
+            final StringBuilder stringBuilder = new StringBuilder();
+            stringBuilder.append(contigId);
+            stringBuilder.append("\t");
+            stringBuilder.append(region1.toString());
+            stringBuilder.append("\t");
+            stringBuilder.append(region2.toString());
+            stringBuilder.append("\t");
+            stringBuilder.append(insertedSequence);
+            stringBuilder.append("\t");
+            stringBuilder.append(homology);
+            return stringBuilder.toString();
+        }
+    }
+
+    static class AlignmentRegion {
+
+        final Cigar cigar;
+        final boolean forwardStrand;
+        final SimpleInterval interval;
```

Comment:
There are conceptually two intervals represented here:  the interval on the assembled contig, and the interval on the reference contig.  Naming could be clearer to keep these straight:  maybe call this `referenceInterval` and change `startInContig` and `endInContig` to `startInAssembledContig` and `endInAssembledContig`?  Or something like that.